src/OneExplorer/OneExplorer.ts

```diff
@@ -80,17 +108,22 @@ enum NodeType {
   product,
 }
 
-class Node {
-  type: NodeType;
+abstract class Node {
+  abstract readonly type: NodeType;
   childNodes: Node[];
   uri: vscode.Uri;
 
-  constructor(type: NodeType, childNodes: Node[], uri: vscode.Uri) {
-    this.type = type;
-    this.childNodes = childNodes;
+  abstract icon: vscode.ThemeIcon;
+  abstract openViewType: string|undefined;
```

Comment:
If openViewType is undefined, it means openView is not supported. Nothing will happen when user clicks the item (no view opened)